### ISSUE TITLE
v0.15.0-alpha.1 chart can't bind socket 80

### ISSUE THREAD

**Issue by vchirikov** [haproxy-ingress/charts#63]
Hi, seems like default chart parameters doesn't work from non-root user:

```
cannot bind socket (Permission denied) for [0.0.0.0:80]
```

I tried to override http port but got in the haproxy.cfg `:80` port:

```cfg
  # # # # # # # # # # # # # # # # # # #
# #
#     HTTP frontend
#
frontend _front_http
    mode http
    bind :80
```

Config:

```yaml
controller:
  containerPorts:
    http: 8080
    https: 4443
  service:
    httpsPorts: []
    httpPorts:
      - port: 8080
        targetPort: 44331
      - port: 4443
        targetPort: http
  config:
    config-sections: |-
      frontend _front_http_redirect
        mode http
        bind :44331        
        # ...
```

**Comment by jcmoraisjr** [comment]
Thanks for the report. Where/how are you deploying the controller? OS flavor and version, kernel version, container runtime flavor and version etc. According to my own tests and a few researches, new kernel and container runtime versions already adds the capability to the process bind to privileged ports, but in fact I'm missing some tests with OS/kernel/cr failing that in order to add proper defaults or proper tips in the getting started doc.

**Comment by jcmoraisjr** [comment]
Hi, can you confirm if you can successfully run the controller adding this config to your `values.yaml`

```yaml
controller:
  podSecurityContext:
    sysctls:
    - name: net.ipv4.ip_unprivileged_port_start
      value: "0"
```

This will config unprivileged port starting at `0`, which means any port, overriding the default `1024`. This will allow any process in the pod to bind to any port number without the need to run as root.

**Comment by jcmoraisjr** [comment]
Fixed, new v0.12~v0.15 releases will change unprivileged ports starting from 1 by default.

**Comment by jcmoraisjr** [comment]
Needed to revert the security context config via #70, we cannot configure it by default because it doesn't work on hostNetwork pods and might violate cluster policies. Otoh we compiled a number of configuration options on [external haproxy example page](https://haproxy-ingress.github.io/docs/examples/external-haproxy/#a-word-about-security), so folks going to configure it have the whys and the hows when configuring.
